src/main/scala/io/picnicml/doddlemodel/preprocessing/RangeScaler.scala

```diff
@@ -0,0 +1,52 @@
+package io.picnicml.doddlemodel.preprocessing
+
+import breeze.linalg.{*, Axis, max, min}
+import cats.syntax.option._
+import io.picnicml.doddlemodel.data.Feature.FeatureIndex
+import io.picnicml.doddlemodel.data.{Features, RealVector}
+import io.picnicml.doddlemodel.typeclasses.Transformer
+import io.picnicml.doddlemodel.syntax.OptionSyntax._
+
+case class RangeScaler private (private val scale: Option[RealVector],
+                                private val minAdjustment: Option[RealVector],
+                                private val range: (Double, Double),
+                                private val featureIndex: FeatureIndex)
+
+object RangeScaler {
+
+  def apply(range: (Double, Double), featureIndex: FeatureIndex): RangeScaler = {
+    val (lowerBound, upperBound) = range
+    val numNumeric = featureIndex.numerical.columnIndices.length
+    require(numNumeric > 0, "There must be at least 1 numeric column in the given data")
```

Comment:
This is handled differently in [the standard scaler case](https://github.com/picnicml/doddle-model/blob/master/src/main/scala/io/picnicml/doddlemodel/preprocessing/StandardScaler.scala#L42); we don't throw the exception if there are no numeric columns and simply no columns are preprocessed (although we should display a warning in that case which is missing there). Not sure what the better way is.

Reply:
Okay, I'll change this - I think both ways could work, so I guess we just make the behavior consistent across preprocessors for now.

Just making sure, this comment applies to `Binarizer` as well, right? (currently, it throws an exception if there are no numerical features in data)

Reply:
I agree, let's go for consistency for now; we can change this later in all files. Yup, it applies to `Binarizer` as well.